src/reconcile/rec_write.c

```diff
@@ -309,13 +309,11 @@ __rec_write_page_status(WT_SESSION_IMPL *session, WT_RECONCILE *r)
             (F_ISSET(r, WT_REC_HS | WT_REC_IN_MEMORY) || page->type == WT_PAGE_COL_FIX));
 
         /*
-         * We have written the page, but something prevents it from being evicted. If we wrote the
-         * newest versions of updates, the on-disk page may contain records that are newer than what
-         * checkpoint would write. Make sure that checkpoint visits the page and (if necessary)
-         * fixes things up.
+         * We have written the page, but something prevents it from being evicted, and the on-disk
```

Comment:
Does this matter in durable history? Even though the on disk value have newer updates than what checkpoint would write, the checkpoint should still be complete and row back to stable will remove those updates I think.

Reply:
Good catch...

The sync code uses `WT_PAGE_MODIFY.first_dirty_txn` to decide if it can skip pages, so that part of the work is still active, I think. I checked in briefly with @michaelcahill, and while he's not confident we can remove `first_dirty_txn`, he does agree this specific line can go.

Fixed in 016401ee2.